src/app/download/download.component.spec.ts

```diff
@@ -21,13 +23,47 @@ describe('DownloadComponent', () => {
     loader = TestbedHarnessEnvironment.loader(fixture);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    state = TestBed.inject(AdAuthoringWorkflowStateContainer);
   });
 
   it('should create the download component', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should call downloadFile when the download button is clicked', async () => {
+  it('should expect download button to be disabled when both file and landingUrl are not initialized', async () => {
+    const button = await loader.getHarness(MatButtonHarness);
+    const isDisbaled = await button.isDisabled();
+
+    expect(isDisbaled).toBe(true);
+  });
+
+  it('should expect download button to be disabled when landingUrl has a value but file is null', async () => {
+    const button = await loader.getHarness(MatButtonHarness);
+
+    state.setState({landingUrl: 'https://google.com'});
+    const isDisbaled = await button.isDisabled();
```

Comment:
nit: Group setup logic together.
```suggestion
    state.setState({landingUrl: 'https://google.com'});
    const button = await loader.getHarness(MatButtonHarness);
    
    const isDisbaled = await button.isDisabled();
```